### ISSUE TITLE
automatically convert line endings in release script

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#2642]
the release script today always publishes whatever files are on the local disk. That means misconfigured get repositories can change line endings and we can end up publishing files that won't work on Macs. We should add in a step that automatically converts all line endings to Unix style.

**Comment by ilyavolodin** [comment]
I'm not sure about automatically converting line-endings dos2unix only does one file at a time, so you have to pull each file individually into it. But to verify line-endings, we can just run eslint with `linebreak-style: [2, 'unix']`.

**Comment by olegafx** [comment]
@ilyavolodin 
1. you could pass a whole dir to `dos2unix`. Something like:
    ```bash
    find . -type f -print0 | xargs -0 dos2unix
    ```
2. You could use `sed`
3. To run `eslint` your `eslint` bin should work correctly :wink:

**Comment by nzakas** [comment]
There's also this: https://www.npmjs.com/package/linefix

**Comment by gyandeeps** [comment]
We can consume prepublish hook to run a script which can make sure all files have Unix line ending and if not make that change.
Now for the script part we can go with:
* [linefix](https://www.npmjs.com/package/linefix)
* [dos2unix for nodejs](https://www.npmjs.com/package/dos2unix)

**Comment by ilyavolodin** [comment]
@gyandeeps That's not really a problem. Problem is that if you modified something locally and then do `npm publish`, your modifications are not going through git at all, so autocrlf or precommit hook will not be launched. That's why it should be part of the release script, to convert line-endings locally (or at least warn about them) before doing `npm publish`.

**Comment by gyandeeps** [comment]
Thats why i mentioned `prepublish` hook.

**Comment by ilyavolodin** [comment]
Oh, sorry, I should read more carefully. Though you said `precommit` hook. But `prepublish` would do the trick. Didn't know about it.

**Comment by gyandeeps** [comment]
When i say hook its not git hook its an npm hook

```js
{
    "script":{
        "prepublish": "node Makefile.js linefix"
    }
}
```

**Comment by ilyavolodin** [comment]
Ok, I'll shut up and will go quietly read documentation in the corner before replying:-)

**Comment by nzakas** [comment]
@gyandeeps that sounds perfect
